File: Phase - 9 Automation Testing Using Python/Pytest Assessment/TutorialNinja/pages/register_page.py
from selenium.webdriver.common.by import By

class RegisterPage:
    first_name = (By.ID, "input-firstname")
    last_name = (By.ID, "input-lastname")
    phone = (By.ID, "input-telephone")
    email_input = (By.ID, "input-email")
    password_input = (By.ID, "input-password")
    confirm_password_input = (By.ID, "input-confirm")
    privacy_policy = (By.XPATH, "//input[@name = \"agree\"]")
    continue_button = (By.XPATH,"//input[@value='Continue']")
    success_message = (By.XPATH, "//h1[text() = \"Your Account Has Been Created!\"]")
    

File: Phase - 12 Playwright Automation/Assessment/Question4/pages/RegisterPage.ts
import {Locator, Page} from '@playwright/test'

export class RegisterPage {
    
    readonly page: Page;
    readonly firstName: Locator;
    readonly lastName: Locator;
    readonly email: Locator;
    readonly telephone: Locator;
    readonly password: Locator;
    readonly confirm: Locator;
    readonly continueButton: Locator;
    readonly privacyPolicy: Locator;

    constructor(page: Page) {
        this.page = page;
        this.firstName = page.locator("//input[@name = 'firstname']")
        this.lastName = page.locator("//input[@name = 'lastname']") ;
        this.email = page.locator("//input[@name = 'email']");
        this.telephone = page.locator("//input[@name = 'telephone']")
        this.password = page.locator("//input[@name = 'password']")
        this.confirm = page.locator("//input[@name = 'confirm']")
        this.continueButton = page.locator("//input[@value = 'Continue']")
        this.privacyPolicy = page.locator("//input[@value = 1 and @type = 'checkbox']")
    }

    async register(firstName:string, lastName:string, email:string, telephone:string, password:string) {
        await this.firstName.fill(firstName);
        await this.lastName.fill(lastName);
        await this.email.fill(email);
        await this.telephone.fill(telephone);
        await this.password.fill(password);
        await this.confirm.fill(password);
        await this.privacyPolicy.click();
        await this.continueButton.click();
    }
}